src/plugins/discord/models.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+
+// https://discordapp.com/developers/docs/reference#snowflakes
+export type Snowflake = string;
+export const ZeroSnowflake: Snowflake = "0";
+
+export type BotToken = string;
+
+/**
+ * Discord Channels can be of various types, defined below.
+ * See: https://discordapp.com/developers/docs/resources/channel#channel-object-channel-types
+ *
+ */
+export type ChannelType =
+  | "GUILD_TEXT"
+  | "DM"
+  | "GUILD_VOICE"
+  | "GROUP_DM"
+  | "GUILD_CATEGORY"
+  | "GUILD_NEWS"
+  | "GUILD_STORE";
+
+/**
+ * The Discord server returns an Id field of type number to represent
+ * the type of Channel. Here we convert that to a text representation,
+ * also based on: https://discordapp.com/developers/docs/resources/channel#channel-object-channel-types
+ *
+ */
+export function channelTypeFromId(id: number): ChannelType {
+  switch (id) {
+    case 0:
+      return "GUILD_TEXT";
+    case 1:
+      return "DM";
+    case 2:
+      return "GUILD_VOICE";
+    case 3:
+      return "GROUP_DM";
+    case 4:
+      return "GUILD_CATEGORY";
+    case 5:
+      return "GUILD_NEWS";
+    case 6:
+      return "GUILD_STORE";
+    default: {
+      throw new Error(`Unknown channel type ID: ${id}`);
+    }
+  }
+}
+
+export type Guild = {|
+  +id: Snowflake,
+  +name: string,
+  +permissions: number,
```

Comment:
Good Q. I'm not sure. @Beanow ?

Reply:
Because we require permissions :smile:

https://github.com/sourcecred/sourcecred/blob/discord/src/plugins/discord/Readme.md

> Permissions bot need:
>
> -  View Channels
> -  Read Message History
>
> `66560` as integer